src/kubernetes.cc

```diff
@@ -823,12 +822,26 @@ void KubernetesReader::WatchMaster(
       if (verbose) {
         LOG(INFO) << "Resetting and running service";
       }
+      if (watch_service->stopped()) {
+        watch_service->reset();
+      }
+      std::unique_ptr<WaitableTimer> timer;
+      if (config_.KubernetesUpdaterWatchReconnectIntervalSeconds() > 0) {
```

Comment:
store `config_.KubernetesUpdaterWatchReconnectIntervalSeconds()` in temp variable, as this value may change between [line 829](https://github.com/Stackdriver/metadata-agent/pull/207/files#diff-3db23052da2a734668d8cbaa14fac121R829) and [line 832](https://github.com/Stackdriver/metadata-agent/pull/207/files#diff-3db23052da2a734668d8cbaa14fac121R832)

Reply:
Then I wonder why we have mutex inside class Configuration because they are not needed. But it is not blocker for this PR.

Reply:
We have the mutex exactly because private methods may be called (for testing, etc). They are just not called here.